repose-aggregator/components/filters/body-patcher-filter/src/main/scala/org/openrepose/filters/bodypatcher/BodyPatcherFilter.scala

```diff
@@ -0,0 +1,150 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.bodypatcher
+
+import java.io.ByteArrayInputStream
+import java.net.URL
+import javax.inject.{Inject, Named}
+import javax.servlet._
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.core.filter.AbstractConfiguredFilter
+import gnieh.diffson.playJson._
+import org.openrepose.commons.utils.io.stream.ServletInputStreamWrapper
+import org.openrepose.commons.utils.servlet.http.ResponseMode.{MUTABLE, PASSTHROUGH}
+import org.openrepose.commons.utils.servlet.http.{HttpServletRequestWrapper, HttpServletResponseWrapper, ResponseMode}
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.filters.bodypatcher.config.ChangeDetails
+import org.openrepose.filters.bodypatcher.config.BodyPatcherConfig
+import org.openrepose.filters.bodypatcher.config.Patch
+import play.api.libs.json.{JsValue, Json => PJson}
+
+import scala.collection.JavaConverters._
+
+/**
+  * Created by adrian on 4/29/16.
+  */
+@Named
+class BodyPatcherFilter @Inject()(configurationService: ConfigurationService)
+  extends AbstractConfiguredFilter[BodyPatcherConfig](configurationService) with LazyLogging {
+  override val DEFAULT_CONFIG: String = "body-patcher.cfg.xml"
+  override val SCHEMA_LOCATION: String = "/META-INF/schema/config/body-patcher.xsd"
+
+  override def doWork(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
+    val httpRequest: HttpServletRequest = request.asInstanceOf[HttpServletRequest]
+    val httpResponse: HttpServletResponse = response.asInstanceOf[HttpServletResponse]
```

Comment:
I'm aware, but i prefer it, because it makes reading it clearer when i the reader know the types. Code is about more than making the compiler happy, we have to maintain it.